### PR TITLE
Migrate to golioth-firmware-sdk

### PR DESCRIPTION
**Pull request by cdwilson**
- Merge changes from [`golioth/reference-design-template@813248f`](https://github.com/golioth/reference-design-template/tree/813248f9a0b1a882140f3906956b34f8dfee0e8a) (this includes updating the `golioth-firmware-sdk` to `v0.11.1`)
- Adds support for the Aludel Elixir Rev A/B boards
- Temporary changes to prevent sending excessive warning log messages on flaky connections

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,9 +17,7 @@ target_sources(app PRIVATE src/main.c)
 target_sources(app PRIVATE src/app_rpc.c)
 target_sources(app PRIVATE src/app_settings.c)
 target_sources(app PRIVATE src/app_state.c)
-target_sources(app PRIVATE src/app_work.c)
-target_sources(app PRIVATE src/dfu/app_dfu.c)
-target_sources_ifdef(CONFIG_BOOTLOADER_MCUBOOT app PRIVATE src/dfu/flash.c)
+target_sources(app PRIVATE src/app_sensors.c)
 target_sources(app PRIVATE src/sensors.c)
```
Comment (cdwilson) [inline CMakeLists.txt:21]: @cdwilson refactor this into separate files for each sensor

File: src/app_rpc.c [hunk 2]
```diff
@@ -137,39 +135,27 @@ static void rpc_log_if_register_failure(int err)
...
 
-	err = golioth_rpc_register(rpc_client, "clean_pm_sensor", on_clean_pm_sensor, NULL);
+	err = golioth_rpc_register(rpc, "clean_pm_sensor", on_clean_pm_sensor, NULL);
 	rpc_log_if_register_failure(err);
 
-	err = golioth_rpc_register(rpc_client, "reset_pm_sensor", on_reset_pm_sensor, NULL);
+	err = golioth_rpc_register(rpc, "reset_pm_sensor", on_reset_pm_sensor, NULL);
 	rpc_log_if_register_failure(err);
 
 	return err;
```
Comment (cdwilson) [inline src/app_rpc.c:161]: @cdwilson this `err` logic inherited from the template doesn't work

File: src/app_sensors.c [hunk 3]
```diff
@@ -32,26 +34,29 @@ static struct sps30_sensor_measurement sps30_sm;
 #define JSON_FMT "{\"tem\":%f,\"pre\":%f,\"hum\":%f,\"co2\":%u,\"mc_1p0\":%f,\"mc_2p5\":%f,\"mc_4p0\":%f,\"mc_10p0\":%f,\"nc_0p5\":%f,\"nc_1p0\":%f,\"nc_2p5\":%f,\"nc_4p0\":%f,\"nc_10p0\":%f,\"tps\":%f}"
```
Comment (cdwilson) [inline src/app_sensors.c:34]: @cdwilson "pretty-print" format this so it's more readable

File: src/app_settings.c [hunk 4]
```diff
@@ -80,177 +81,127 @@ static void sps30_sensor_set_fan_auto_cleaning_interval_work_handler(struct k_wo
...
-			k_work_submit(&scd4x_sensor_set_temperature_offset_work);
-		}
-		return GOLIOTH_SETTINGS_SUCCESS;
-	}
+static enum golioth_settings_status on_sps30_cleaning_interval_setting(int32_t new_value, void *arg)
+{
+	_sps30_cleaning_interval_s = (uint32_t)new_value;
+	LOG_INF("Set SPS30 cleaning interval to %i seconds", _sps30_cleaning_interval_s);
+	return GOLIOTH_SETTINGS_SUCCESS;
+}
```
Comment (cdwilson) [inline src/app_settings.c:133]:
```suggestion
static enum golioth_settings_status on_sps30_cleaning_interval_setting(int32_t new_value, void *arg)
{
	_sps30_cleaning_interval_s = (uint32_t)new_value;
	LOG_INF("Set SPS30 cleaning interval to %i seconds", _sps30_cleaning_interval_s);
	/* Submit a work item to write this setting to the sensor */
	k_work_submit(&sps30_sensor_set_fan_auto_cleaning_interval_work);
	return GOLIOTH_SETTINGS_SUCCESS;
}
```

**Review by szczys — APPROVED** [review]

**Pull request merged**